api/send_request.php

```diff
@@ -0,0 +1,264 @@
+$address = mysqli_real_escape_string($mysqli, strip_tags(trim($_POST['address'])));
+$receipt = $_FILES['receipt'];
+$docs = $_FILES['docs'];
+
+$receipt_mime_types = array(
+    'pdf'  => 'application/pdf',
+    'doc'  => 'application/msword',
+    'docx' => 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    'png'  => 'image/png',
+    'jpeg' => 'image/jpeg',
+    'jpg'  => 'image/jpeg'
+);
+
+$docs_mime_types = array(
+    'pdf'  => 'application/pdf',
+    'doc'  => 'application/msword',
+    'docx' => 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    'png'  => 'image/png',
+    'jpeg' => 'image/jpeg',
+    'jpg'  => 'image/jpeg'
+);
+
+// Initialize data validation and SQL result
+if (!isset($result_data))
+    $result_data = new stdClass();
+$result_data->status = 'error';
+$result_data->message = '';
+
+// Check name
+if (!preg_match("/^[\w\ \'\.]{1,128}$/", $name)) {
+    $result_data->message = 'Your name is invalid. Please only use latin characters a-z with an optional '
+        . 'apostrophe or period. Your name is also limited to 128 characters.';
+    echo json_encode($result_data);
+    die();
+}
+
+// Check position
+if (!preg_match("/^[\w\ \'\.]{1,128}$/", $position)) {
+    $result_data->message = 'Your position is invalid. Please only use latin characters a-z with an optional '
+        . 'apostrophe or period. Your position is also limited to 128 characters.';
+    echo json_encode($result_data);
+    die();
+}
+
+// Check email
+if(!preg_match('/^[\w\W]+@[\w\W\d]{1,128}$/', $email)) {
+    $result_data->message = 'Your email is invalid. Please use an email in the following format: <>@<>. '
+        . 'Your email is also limited to 128 characters.';
+    echo json_encode($result_data);
+    die();
+}
+
+// Check mId
+if (!preg_match("/^M[0-9]{8}$/", $mId)) {
+    $result_data->message = 'Your M# is invalid. Please only use official format (Mxxxxxxxx) with an optional '
+        . 'apostrophe or period.';
+    echo json_encode($result_data);
+    die();
+}
+
+// Check date
+if(!preg_match('/^(19|20)\d\d([-])(0[1-9]|1[012])\2(0[1-9]|[12][0-9]|3[01])$/', $date)) {
+    $result_data->message = 'Your expenditure date is invalid. Please format it in the following way: yyyy/mm/dd.';
+    echo json_encode($result_data);
+    die();
+}
+
+// Check vendor
+if (!preg_match("/^[\w\ \'\.]{1,128}$/", $vendor)) {
+    $result_data->message = 'Your vendor name is invalid. Please only use latin characters a-z with an optional '
+        . 'apostrophe or period. Your vendor is also limited to 128 characters.';
+    echo json_encode($result_data);
+    die();
+}
+
+// Check amount
+if (!preg_match("/^[0-9]{0,10}.[0-9]{0,10}$/", $amount)) {
+    $result_data->message = 'Your amount is invalid. Please only use only numerical characters';
+    echo json_encode($result_data);
+    die();
+}
+
+// Check description
+if (!preg_match("/^[\w\ \'\.]{1,128}$/", $description)) {
```

Comment:
You probably don't want to limit the description to just 128. Perhaps 500 characters is better?